### DISCUSSION TITLE
A macro to create a signal, like perseus's MakeRx

### DISCUSSION THREAD

**Discussion by huang12zheng** [leptos-rs/leptos#1233] [Ideas]
show code:
* input
```rs
#[derive(Serialize, Deserialize, ReactiveState)]
#[rx(alias = "AppStateRx")]
pub struct AppState {
    /// Authentication data accessible to all pages.
    #[rx(nested)]
    pub auth: RxVec<AuthData>,
}
```
* output:
```rs
#[derive(Clone)]
pub struct AppStatePerseusRxIntermediate {
    pub auth: <RxVec<AuthData> as ::perseus::state::MakeRx>::Rx,
}

impl From<AppState> for AppStatePerseusRxIntermediate {
    fn from(value: AppState) -> AppStatePerseusRxIntermediate {
        use ::perseus::state::MakeRx;
        value.make_rx()
    }
}

impl ::perseus::state::MakeRx for AppState {
    type Rx = AppStatePerseusRxIntermediate;
    #[cfg(debug_assertions)]
    const HSR_IGNORE: bool = false;
    fn make_rx(self) -> Self::Rx {
        use ::perseus::state::MakeRx;
        Self::Rx {
            auth: self.auth.make_rx(),
        }
    }
}
...
>> and there are things like RxVec or RxHashMap.


```

I can just add a macro and go from the basic schema to leptos signal. What do you think of this idea?

**Comment by huang12zheng** [comment]
with a code under your branch "effect-cleanups"
https://github.com/huang12zheng/rx-state/blob/099f4344de9633e52565c6ce3cae2972a68b722a/crates/rx-macro/tests/compile.rs
https://github.com/huang12zheng/rx-state/blob/main/crates/rx-macro/src/snapshots/rx_macro__expand_derive.snap
https://github.com/huang12zheng/rx-state/blob/main/crates/rx-macro/src/snapshots/rx_macro__expand_nest.snap

**Comment by gbj** [comment]
Yes, I think this kind of wrapper is very convenient. I have wondered what the best way is to replicate something like Solid's "stores" (which give reactive access to deeply-nested form fields) and this is certainly one approach.
